workers/generic-worker/mounts.go

```diff
@@ -756,50 +810,79 @@ func UnmarshalInto(c json.RawMessage, fsContent FSContent) (FSContent, error) {
 // Downloads ArtifactContent to a file inside the downloads directory specified
 // in the global config file. The filename is a random slugid, and the
 // absolute path of the file is returned.
-func (ac *ArtifactContent) Download(task *TaskRun) (file string, sha256 string, err error) {
+func (ac *ArtifactContent) Download(taskMount *TaskMount) (file string, sha256 string, err error) {
 	basename := slugid.Nice()
 	file = filepath.Join(config.DownloadsDir, basename)
 
-	task.Infof("[mounts] Downloading %v to %v", ac, file)
+	taskMount.Infof("Downloading %v to %v", ac, file)
 
 	var runID int64 = -1 // use the latest run
-	_, contentLength, err := task.Queue.DownloadArtifactToFile(ac.TaskID, runID, ac.Artifact, file)
+	_, contentLength, err := taskMount.task.Queue.DownloadArtifactToFile(ac.TaskID, runID, ac.Artifact, file)
 	if err != nil {
 		return
 	}
 
 	sha256, err = fileutil.CalculateSHA256(file)
 	if err != nil {
-		task.Infof("[mounts] Downloaded %v bytes from %s to %v but cannot calculate SHA256", contentLength, ac, file)
+		taskMount.Infof("Downloaded %v bytes from %s to %v but cannot calculate SHA256", contentLength, ac, file)
 		panic(fmt.Sprintf("Internal worker bug! Cannot calculate SHA256 of file %v that I just downloaded: %v", file, err))
 	}
-	task.Infof("[mounts] Downloaded %v bytes with SHA256 %v from %s to %v", contentLength, sha256, ac, file)
+	taskMount.Infof("Downloaded %v bytes with SHA256 %v from %s to %v", contentLength, sha256, ac, file)
 	return
 }
 
+func (ic *IndexedContent) Download(taskMount *TaskMount) (file string, sha256 string, err error) {
+	itr, err := taskMount.index.FindTask(ic.Namespace)
+	if err != nil {
+		return "", "", err
+	}
+	ac := &ArtifactContent{
+		Artifact: ic.Artifact,
+		Sha256:   "",
```

Comment:
Should we calculate this and set it?

Reply:
The mounts feature calculates the value, and compares it to what is passed in here, so if we did calculate it, the two values would match, but there would be a little bit more overhead. If we don't set it, the value isn't checked. Note, I could have just not included it in the struct literal, because the zero value would be the empty string, but I just included it here to be a bit more explicit, so that it is clear that it exists, but isn't used.